### PR TITLE
Feat/i18n

### LINKED ISSUE #1555: Guidance for localization/internationalization patterns
### Is this issue related to the design system core library or design system website?

design-system-core

### Expected Behavior

Applications frequently have a requirement to be localized to many different languages.  This is usually achieved by creating a bundle of messages in each locale, and loading the correct locale's bundle at run/render time.

For user interfaces specifically the challenges usually revolve around binding dynamic values into localized text, and situations where markup and localized text needs to be mixed.

I've included a screenshot and snippet of some common cases of text that may need to be localized (the scenarios can be combined, such as dynamic values + markup in the same string).

### Actual Behavior

Since markup in react is achieved via components, it's not clear what a good strategy would be for i18n/l10n.  Can you provide guidance on recommended patterns for implementing i18n/l10n using grommet/design system?

### URL, screen shot, or CodeSandbox exhibiting the issue

```jsx
<Box direction="column">
  <Text>{5} of {10} items</Text>
  <Text>
    mixing <Text weight="bold">markup</Text> with text
  </Text>
  <Text>{/* date time formatting */}    
    {'04/15/2021 12:45 PM CST'}
  </Text>
</Box>
```

image

### PR DESCRIPTION
**Pull request by MikeKingdom**
#### What does this PR do?

Add guidance for Internationalization

#### Where should the reviewer start?

internationalization.mdx

#### What are the relevant issues?
Closes #1555

### CODE REVIEW

**Review by ericsoderberghp — APPROVED** [review]
I'm good. But, it would be good to get Taylor and Matt's guidance editing eyes on it too.

File: aries-site/src/pages/templates/internationalization.mdx [hunk 1]
```diff
@@ -0,0 +1,97 @@
+
+## Overview
+
+Internationalizion (I18N) and Localization (L10N) are important for web content
+and applications which are intended to be used by global users. It's even important
+for users in a single location who simply speak different languages.
+
+Internationalization is the process of making your application tailor its
```
Comment (taysea) [inline aries-site/src/pages/templates/internationalization.mdx:8]:
```suggestion
Internationalization is the process of tailoring application
```

File: aries-site/src/pages/templates/internationalization.mdx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+different language, users can benefit from your site being internationalized. Part
+of internationalizating is making your site recognize the locale set in the browser
+and format things like dates and times, currency and numbers specifically
+for that locale. Browsers have great built-in support for formatting that content
+appropriately for the locale. For example, a user in Australia or the United Kingdom
+may be perfectly happy with your content being in English but would much prefer any
+dates or times be presented with the day of the month before the month
+(28/06/2021 versus 6/28/2021). This is easy to do with `Date.toLocalDateString()`.
+More information on browser and javsascript localization support is available from
+many sources such as ***TODO***
```
Comment (taysea) [inline aries-site/src/pages/templates/internationalization.mdx:22]: If we don't plan to fill this "TODO" in on the first pass, we should remove it

File: aries-site/src/pages/templates/internationalization.mdx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+## Guidelines
+
+Depending on how your site or application is implemented, you'll need to pick
+an appropriate mechanism for choosing a language and locale and then provide
+content for the language and locale the user chose. You can provide a way to pick
```
Comment (taysea) [inline aries-site/src/pages/templates/internationalization.mdx:28]:
```suggestion
content for the language and locale the user chose. You can provide a control in the application that allows the user to pick
```

File: aries-site/src/pages/templates/internationalization.mdx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+a language or simply try to use the preference set by the browser. Browsers include
```
Comment (taysea) [inline aries-site/src/pages/templates/internationalization.mdx:29]: Maybe rewording this section to something like, "When possible, set localize content based on the locale preferences set by the user or returned in the header of the page request."

File: aries-site/src/pages/templates/internationalization.mdx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+the user's locale preferences (if set) in header in the page request. Your
+application or site can choose localized content that best matches a locale from
+this list. Even if you don't have content for one of the preferred languages or
+allowed the user to set the language another way, it's best to still format
+dates, numbers and such given their preferred locale whenever possible.
+
+Once the site has chosen a locale from the user's preferences your site needs to
+provide that content. If you used Grommet to implement the site, there are some
```
Comment (taysea) [inline aries-site/src/pages/templates/internationalization.mdx:37]:
```suggestion
provide the appropriate content given the selected locale. If you have implemented your application with Grommet, there are some default messages for components like Menu and FormField that will require localization. For example, the required message that accompanies a required FormField.
```

File: aries-site/src/pages/templates/internationalization.mdx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+messages used in Grommet components that you'll need to localize also.
```
Comment (taysea) [inline aries-site/src/pages/templates/internationalization.mdx:38]:
```suggestion
```

File: aries-site/src/pages/templates/internationalization.mdx [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+
+
+## Localizing Grommet
+
+Messages needed for Grommet components can be provided all up front through the
```
Comment (taysea) [inline aries-site/src/pages/templates/internationalization.mdx:43]:
```suggestion
Messages needed for Grommet components can be provided in a single place by using the
```

File: aries-site/src/data/structures/templates.js [hunk 8]
```diff
@@ -251,4 +251,23 @@ export const templates = [
     sections: [],
     relatedContent: ['Table', 'Lists', 'Filtering'],
   },
+  {
+    name: 'Internationalization',
+    available: false,
+    description: 'Internationalizing a Design System site',
```
Comment (taysea) [inline aries-site/src/data/structures/templates.js:257]:
```suggestion
    description: `How to internationalize a site or application that uses
    the HPE Design System`,
```

File: aries-site/src/data/structures/templates.js [hunk 9]
```diff
@@ -251,4 +251,23 @@ export const templates = [
     sections: [],
     relatedContent: ['Table', 'Lists', 'Filtering'],
   },
+  {
+    name: 'Internationalization',
+    available: false,
```
Comment (taysea) [inline aries-site/src/data/structures/templates.js:256]:
```suggestion
    available: true,
```
Should this be true?

Comment (MikeKingdom) [reply aries-site/src/data/structures/templates.js:256]: Good question. I wasn't quite sure why a couple spots sort by this. Thoughts?

File: aries-site/src/pages/templates/internationalization.mdx (hunk 1 above)
Comment (MikeKingdom) [reply aries-site/src/pages/templates/internationalization.mdx:8]: Thanks @taysea ! I commited most of the wording tweaks.

File: aries-site/src/data/structures/templates.js (hunk 9 above)
Comment (halocline) [reply aries-site/src/data/structures/templates.js]: `available` property is used in to indicate we have Design System styling and guidance. When sorting, it is used to more prominently feature components/templates cards for which we do have styling and guidance.

This designation was relevant in the early days of the Design System, but as the coverage across components has increased, this property is less relevant and can likely be deprecated in the near future.

**Comment by halocline** [comment]
Apologies that I wasn't able to squeeze this in today. Feel free to get this published.

File: aries-site/src/data/structures/templates.js [hunk 10]
```diff
@@ -263,6 +263,26 @@ export const templates = [
     sections: [],
     relatedContent: ['Table', 'Lists', 'Filtering'],
   },
+  {
+    name: 'Internationalization',
+    available: true,
+    description: `How to internationalize a site or application that uses the 
+    HPE Design System`,
```
Comment (taysea) [inline aries-site/src/data/structures/templates.js:270]:
```suggestion
    HPE Design System.`,
```

File: aries-site/src/pages/templates/internationalization.mdx [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+the user's locale preferences (if set) in header in the page request. Your
+application or site can choose localized content that best matches a locale from
+this list. Even if you don't have content for one of the preferred languages or
+allowed the user to set the language another way, it's best to still format
+dates, numbers and such given their preferred locale whenever possible.
+
+Once the site has chosen a locale from the user's preferences your site needs to
+provide the appropriate content given the selected locale. If you have implemented
+your application with Grommet, there are some default messages for components like
+Menu and FormField that will require localization, For example, the required message
```
Comment (taysea) [inline aries-site/src/pages/templates/internationalization.mdx:38]:
```suggestion
Menu and FormField that will require localization. For example, the required message
```

File: aries-site/src/pages/templates/internationalization.mdx [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+
+Then just provide the Grommet messages along with your other localized messages.
+See the [Grommet component documentation](https://v2.grommet.io/grommet)
+for details on what messages are needed.
+
+If needed you can also provide customized messages to the Grommet components
+which need them. The components which have localizable messages also support
+a `messages` property which accepts their specific messages. In most cases it
+easiest to provide the messages as above to the `Grommet` component instead.
+However if you want to specify customized messages for a component in a few
```
Comment (taysea) [inline aries-site/src/pages/templates/internationalization.mdx:86]:
```suggestion
However, if you want to specify customized messages for a component in a few
```

**Review by taysea — APPROVED** [review]
